### PR TITLE
minimap plugin: add touch events

### PR DESCRIPTION
**Pull request by jnorris441**
The minimap should be draggable on a phone

### CODE REVIEW

File: src/plugin/minimap/index.js [hunk 1]
```diff
@@ -307,35 +307,46 @@ export default class MinimapPlugin {
...
                 this.draggingOverview = true;
-                relativePositionX = event.layerX;
-                positionMouseDown.clientX = event.clientX;
-                positionMouseDown.clientY = event.clientY;
-            });
+                relativePositionX = layerX(event);
+                positionMouseDown.clientX = clientX(event);
+                positionMouseDown.clientY = clientY(event);
+            };
+            this.overviewRegion.addEventListener('mousedown', mouseDown);
```
Comment (thijstriemstra) [inline src/plugin/minimap/index.js:320]: these event listeners should be cleaned up as well.

File: src/plugin/minimap/index.js [hunk 2]
```diff
@@ -307,35 +307,46 @@ export default class MinimapPlugin {
...
                     this.moveOverviewRegion(
-                        event.clientX -
+                        clientX(event) -
                             this.drawer.container.getBoundingClientRect().left -
                             relativePositionX
                     );
                 }
-            });
+            };
+            this.drawer.wrapper.addEventListener('mousemove', mouseMove);
```
Comment (thijstriemstra) [inline src/plugin/minimap/index.js:332]: these event listeners should be cleaned up as well.

File: src/plugin/minimap/index.js [hunk 3]
```diff
@@ -307,35 +307,46 @@ export default class MinimapPlugin {
...
                 ) {
                     seek = true;
                     this.draggingOverview = false;
                 } else if (this.draggingOverview) {
                     seek = false;
                     this.draggingOverview = false;
                 }
-            });
+            };
+            this.drawer.wrapper.addEventListener('mouseup', mouseUp);
```
Comment (thijstriemstra) [inline src/plugin/minimap/index.js:347]: these event listeners should be cleaned up as well.

**Review by thijstriemstra — CHANGES_REQUESTED** [review]
can you also add a changelog entry?

File: src/plugin/minimap/index.js (hunk 1 above)
Comment (jnorris441) [reply src/plugin/minimap/index.js:320]: Makes sense. What is the best way to do that? How are the existing mousedown/mousemove/mouseup cleaned up?

Comment (thijstriemstra) [reply src/plugin/minimap/index.js:320]: take a look at other plugins like regions or timeline.

Comment (thijstriemstra) [reply src/plugin/minimap/index.js:320]: and others in `destroy`: https://github.com/katspaugh/wavesurfer.js/blob/bfa9c76040058ac59bab19e7bd9d347521c6e425/src/plugin/minimap/index.js#L181

**Comment by thijstriemstra** [comment]
@jnorris441 can you take a look at this PR this week, I'd like to release a new wavesurfer.js with this fix included if possible.

**Comment by thijstriemstra** [comment]
no feedback in a year, closing.

**Pull request closed without merging**